semanticdb/metap/src/main/scala/scala/meta/cli/Metap.scala

```diff
@@ -120,6 +122,204 @@ object Metap {
     }
   }
 
+  private val symCache = new WeakHashMap[TextDocument, immutable.Map[String, SymbolInformation]]
```

Comment:
What is the hashcode impl for TextDocument?

Reply:
ScalaPB messages are case classes, so we're kind of fine (as in protected against duplicate uris). It would be ideal if there was `WeakIdentityHashMap`, but it's not there, and I think that implementing one is not worth it.